### PR TITLE
Porting Intake to Java and Adding to Constants.java

### PR DESCRIPTION
**Pull request by hpark-24**

### CODE REVIEW

**Review by jmullins16 — COMMENTED** [review]
Can we change the "Deployer" references (method names AND text writting to log) to be "Arm" or "arm"? Except for the state put to the dashboard for "IN_Deployed". I think the code will be much more understandable if people recognize that those references are all to the "arm" object.

File: Offseason2022/src/main/java/frc/robot/subsystems/Intake.java [hunk 1]
```diff
@@ -4,27 +4,105 @@
...
-    arm = new Solenoid(0, PneumaticsModuleType.CTREPCM, 0);
+    setName("Intake");
+    setSubsystem("Intake");
     addChild("Arm", arm);
+
+    SmartDashboard.putBoolean("HL_IN8Valid", talonValidIN8);
+
+    // Check if solenoids are functional or blacklisted
+    if (arm.isDisabled( ))
+      DataLogManager.log(getSubsystem( ) + "Deploy Solenoid is BLACKLISTED");
```
Comment (reevar24) [inline Offseason2022/src/main/java/frc/robot/subsystems/Intake.java:40]: Looks good! One suggestion though, you might need to put colons after the getSubsystem( ) calls.

**Review by jmullins16 — CHANGES_REQUESTED** [review]
For the data log statements,
- The getSubsystem call will print the name of the subsystem, so we can skip the "IN"
- The squiggly brackets are a C++ field marker that tells spdlog where to put the following parameters. DattLogManager treats everything like a string and concatenates them--so the brackets aren't used.
- decisions can be put into the log strings so only one statement is needed to handle both cases

      DataLogManager.log(getSubsystem( ) + ": Arm " + ((extended) ? "DEPLOYED" : "STOWED"));

**Review by jmullins16 — CHANGES_REQUESTED** [review]
1) IntakeRun: m_intakeDirection naming - this was not a very good choice last year, and we have a chance to make it better. The Intake subsystem has two properties--the arm has a position, and the motor has a speed (in the constants file we call this a Mode). So let's make this variable name more closely related to the what it holds--a Mode. So maybe m_mode would be a good choice.
2) In IntakeRun we can eliminate the DataLog calls in Init and Interrupted. We have code in RobotInit that prints out the init and end messsages, so delete those two lines.
3) Intake subsystem: Better shorthand for the pneumatics arm.isDisabled check on one line:
      DataLogManager.log(getSubsystem( ) + ": Arm Solenoid is " + ((arm.isDisabled) ? "BLACKLISTED" : "OK"));
4) The code autoformatter doesn't remove blank lines right after an opening bracket and right before a closing bracket. So check for any and remove them as a final cleanup.
Functionally, I think this all works and be fine in the robot. Just clean up some of the style issues, and it's good to go. Nice work!

File: Offseason2022/src/main/java/frc/robot/commands/IntakeRun.java [hunk 2]
```diff
@@ -14,12 +14,12 @@
 public class IntakeRun extends CommandBase
 {
   private Intake m_intake;
-  private Mode   m_intakeDirection;
+  private Mode   m_mode;
 
   public void intakeRun(Intake intake, Mode intakeDirection)
```
Comment (jmullins16) [inline Offseason2022/src/main/java/frc/robot/commands/IntakeRun.java:19]: Looks great except for I don't think it compiles yet. For the constructor intakeRun, it has a parameter intakeDirection, but in the body of the method you've changed it (correctly) to be "mode".

Once it compiles with that change, let's have at least one other person review and get this merged.

**Comment by jmullins16** [comment]
Looks correct now and ready for merge.

**Pull request merged**